lib/dialects/abstract/query-generator.js

```diff
@@ -300,10 +300,20 @@ class QueryGenerator {
     }
 
     if (this._dialect.supports.inserts.updateOnDuplicate && options.updateOnDuplicate) {
-      onDuplicateKeyUpdate = ` ON DUPLICATE KEY UPDATE ${options.updateOnDuplicate.map(attr => {
-        const key = this.quoteIdentifier(attr);
-        return `${key}=VALUES(${key})`;
-      }).join(',')}`;
+      if (this._dialect.supports.inserts.updateOnDuplicate == ' ON CONFLICT DO UPDATE SET') { // postgres
+        // If no conflict target columns were specified, use the primary key names from options.upsertKeys
+        onDuplicateKeyUpdate = ` ON CONFLICT (${options.upsertKeys.map(attr => {
+          return `"${attr}"`;
```

Comment:
Also use `this.quoteIdentifier(<attr>)` instead of `"<attr>"`

Reply:
In commit 8381105 I updated with `quoteIdentifier()` (oops). 

I am not sure how best to inline this though? Especially since I added the `quoteIdentifier()` call? My JS-fu is weak, there is probably some classic syntax I am not familiar with. I was following the existing code example which looks like this:

```
onDuplicateKeyUpdate = `${this._dialect.supports.inserts.updateOnDuplicate} ${options.updateOnDuplicate.map(attr => {
  const key = this.quoteIdentifier(attr);
  return `${key}=VALUES(${key})`;
}).join(',')}`;
```